File: Kotlin/src/main/java/no/agitec/fagaften/mars/kotlin/section3/learnprograming/bitoperatorcasting/Declaration.java
package no.agitec.fagaften.mars.kotlin.section3.learnprograming.bitoperatorcasting;

import no.agitec.fagaften.mars.kotlin.section3.learnprograming.equality.Employee;
import no.agitec.fagaften.mars.kotlin.section3.learnprograming.equality.Employeej;

public class Declaration {


    public static void main(String ... args) {


        int x = 0x00101101;
        int y = 0x11011011;
        int a = x | y;
        int b = x & y;
        int c = x ^ y;


        Employeej employeeOne = new Employeej("Mary", 1);
        Employeej employeeFour = new Employeej("John", 2);

        Object something = employeeFour;
        if (something instanceof  Employee) {
            //val newEmployee = something as Employee
            Employeej somethingj = (Employeej) employeeOne;
            System.out.println(somethingj.getName());
        }

    }


}

File: Kotlin/src/main/java/no/agitec/fagaften/mars/kotlin/section3/learnprograming/bitoperatorcasting/Declaration.kt
package no.agitec.fagaften.mars.kotlin.section3.learnprograming.bitoperatorcasting

import no.agitec.fagaften.mars.kotlin.section3.learnprograming.equality.Employee

fun main(args: Array<String>)  {

    //bit operator
    val x = 0x00101101
    val y = 0x11011011
    val a = x or y
    val b = x and y
    val c = x xor y


    //Casting
    val employeeOne = Employee("Mary", 1)
    val employeeFour = Employee("John", 2)

    var something: Any = employeeFour
    if (something is Employee) { //instanceof
        //val newEmployee = something as Employee
        something = employeeFour //Only Employee
        println(something.name) //John

        something = "notEmployee" //Kotlin know it is a error here. Only allow Employee
        //println(something.name) //Error:(26, 27) Kotlin: Unresolved reference: name

        something = employeeOne
        println(something.name) //Mary
    }




}
